File: crates/alith/examples/agent_deepseek.rs
use alith::{Agent, LLM};

#[tokio::main]
async fn main() -> Result<(), anyhow::Error> {
    let model = LLM::openai_compatible_model(
        "<Your API Key>", // Replace with your api key or read it from env.
        "api.deepseek.com",
        "deepseek-chat", // or `deepseek-reasoner` for DeepSeek R1
    )?;
    let agent = Agent::new("simple agent", model, vec![])
        .preamble("You are a comedian here to entertain the user using humour and jokes.");
    let response = agent.prompt("Entertain me!").await?;

    println!("{}", response);

    Ok(())
}

File: sdks/node/examples/agent_deepseek.ts
import { Agent } from 'alith'

const agent = new Agent({
  name: 'A dummy Agent',
  model: 'deepseek-chat', // or `deepseek-reasoner` for DeepSeek R1
  apiKey: '<Your API Key>',
  baseUrl: 'api.deepseek.com',
  preamble:
    'You are a calculator here to help the user perform arithmetic operations. Use the tools provided to answer the user question.',
})
console.log(agent.prompt('Calculate 10 - 3'))
